packages/web/src/components/list/MultiDataList.js

```diff
@@ -214,6 +247,41 @@ class MultiDataList extends Component {
 		});
 	};
 
+	static generateQueryOptions(props, state) {
+		const queryOptions = getQueryOptions(props);
+		const includes = state.options.map(item => item.value);
+		return getAggsQuery(queryOptions, props, includes);
+	}
+
+	updateQueryOptions = (props) => {
+		const queryOptions = MultiDataList.generateQueryOptions(props, this.state);
+		props.setQueryOptions(this.internalComponent, queryOptions);
+	};
+
+	updateStateOptions = (bucket) => {
+		if (bucket) {
+			const { options } = this.state;
+			const newOptions = [];
+
+			options.forEach((item) => {
+				const doc = bucket.find(bucketItem => bucketItem.key === item.value);
```

Comment:
This looks performance intensive. Can we improve this a little bit? `find` will run on every iteration. Maybe we can flatten the buckets into a dictionary to bring the traversal complexity to O(1).

Reply:
Also, I'd recommend you to use `map` here instead of `forEach` and `push`, it will improve the readability further 😸